### ISSUE TITLE
Unable to export XLSX with more than 64k cells

### ISSUE THREAD

**Issue by d4rkstar** [metabase/metabase#13271]
**Describe the bug**
After the update from 0.35 to 0.36.6 i'm unable to export XLSX if the report includes type/DateTime (timestamp without time zone) columns. The same report, without date columns, will exports without problem.

**Logs**
```
metabase_1_c425448d3056 |   :params nil},
metabase_1_c425448d3056 |  :status :failed,
metabase_1_c425448d3056 |  :class java.lang.IllegalStateException,
metabase_1_c425448d3056 |  :stacktrace
metabase_1_c425448d3056 |  ["org.apache.poi.xssf.model.StylesTable.createCellStyle(StylesTable.java:815)"
metabase_1_c425448d3056 |   "org.apache.poi.xssf.usermodel.XSSFWorkbook.createCellStyle(XSSFWorkbook.java:730)"
metabase_1_c425448d3056 |   "org.apache.poi.xssf.streaming.SXSSFWorkbook.createCellStyle(SXSSFWorkbook.java:858)"
metabase_1_c425448d3056 |   "dk.ative.docjure.spreadsheet$apply_date_format_BANG_.invokeStatic(spreadsheet.clj:225)"
metabase_1_c425448d3056 |   "dk.ative.docjure.spreadsheet$apply_date_format_BANG_.invoke(spreadsheet.clj:223)"
metabase_1_c425448d3056 |   "dk.ative.docjure.spreadsheet$fn__35183.invokeStatic(spreadsheet.clj:252)"
metabase_1_c425448d3056 |   "dk.ative.docjure.spreadsheet$fn__35183.invoke(spreadsheet.clj:248)"
metabase_1_c425448d3056 |   "clojure.lang.MultiFn.invoke(MultiFn.java:234)"
metabase_1_c425448d3056 |   "dk.ative.docjure.spreadsheet$add_row_BANG_.invokeStatic(spreadsheet.clj:267)"
metabase_1_c425448d3056 |   "dk.ative.docjure.spreadsheet$add_row_BANG_.invoke(spreadsheet.clj:260)"
metabase_1_c425448d3056 |   "--> query_processor.streaming.xlsx$fn$reify__35434.write_row_BANG_(xlsx.clj:76)"
metabase_1_c425448d3056 |   "query_processor.streaming$streaming_rff$fn__35441$fn__35442.invoke(streaming.clj:35)"
metabase_1_c425448d3056 |   "query_processor.middleware.process_userland_query$add_and_save_execution_info_xform_BANG_$execution_info_rf_STAR___46380.invoke(process_userland_query.clj:95)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_rows_truncated$add_rows_truncated_xform$fn__39275.invoke(add_rows_truncated.clj:28)"
metabase_1_c425448d3056 |   "query_processor.middleware.limit$limit_xform$fn__45618.invoke(limit.clj:25)"
metabase_1_c425448d3056 |   "query_processor.middleware.large_int_id$result_int__GT_string$fn__45589.invoke(large_int_id.clj:12)"
metabase_1_c425448d3056 |   "query_processor.reducible$reducible_rows$reify__38090.reduce(reducible.clj:157)"
metabase_1_c425448d3056 |   "query_processor.context.default$default_reducef$fn__37988.invoke(default.clj:59)"
metabase_1_c425448d3056 |   "query_processor.context.default$default_reducef.invokeStatic(default.clj:58)"
metabase_1_c425448d3056 |   "query_processor.context.default$default_reducef.invoke(default.clj:49)"
metabase_1_c425448d3056 |   "query_processor.context$reducef.invokeStatic(context.clj:69)"
metabase_1_c425448d3056 |   "query_processor.context$reducef.invoke(context.clj:62)"
metabase_1_c425448d3056 |   "query_processor.context.default$default_runf$respond_STAR___37992.invoke(default.clj:70)"
metabase_1_c425448d3056 |   "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:392)"
metabase_1_c425448d3056 |   "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:377)"
metabase_1_c425448d3056 |   "driver.sql_jdbc$fn__73933.invokeStatic(sql_jdbc.clj:49)"
metabase_1_c425448d3056 |   "driver.sql_jdbc$fn__73933.invoke(sql_jdbc.clj:47)"
metabase_1_c425448d3056 |   "query_processor.context$executef.invokeStatic(context.clj:59)"
metabase_1_c425448d3056 |   "query_processor.context$executef.invoke(context.clj:48)"
metabase_1_c425448d3056 |   "query_processor.context.default$default_runf.invokeStatic(default.clj:69)"
metabase_1_c425448d3056 |   "query_processor.context.default$default_runf.invoke(default.clj:67)"
metabase_1_c425448d3056 |   "query_processor.context$runf.invokeStatic(context.clj:45)"
metabase_1_c425448d3056 |   "query_processor.context$runf.invoke(context.clj:39)"
metabase_1_c425448d3056 |   "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
metabase_1_c425448d3056 |   "query_processor.reducible$pivot.invoke(reducible.clj:31)"
metabase_1_c425448d3056 |   "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__45635.invoke(mbql_to_native.clj:26)"
metabase_1_c425448d3056 |   "query_processor.middleware.check_features$check_features$fn__44911.invoke(check_features.clj:42)"
metabase_1_c425448d3056 |   "query_processor.middleware.optimize_datetime_filters$optimize_datetime_filters$fn__45800.invoke(optimize_datetime_filters.clj:133)"
metabase_1_c425448d3056 |   "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__47328.invoke(wrap_value_literals.clj:137)"
metabase_1_c425448d3056 |   "query_processor.middleware.annotate$add_column_info$fn__43532.invoke(annotate.clj:574)"
metabase_1_c425448d3056 |   "query_processor.middleware.permissions$check_query_permissions$fn__44786.invoke(permissions.clj:64)"
metabase_1_c425448d3056 |   "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__46318.invoke(pre_alias_aggregations.clj:40)"
metabase_1_c425448d3056 |   "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__44984.invoke(cumulative_aggregations.clj:61)"
metabase_1_c425448d3056 |   "query_processor.middleware.resolve_joins$resolve_joins$fn__46850.invoke(resolve_joins.clj:183)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__39262.invoke(add_implicit_joins.clj:245)"
metabase_1_c425448d3056 |   "query_processor.middleware.large_int_id$convert_id_to_string$fn__45596.invoke(large_int_id.clj:44)"
metabase_1_c425448d3056 |   "query_processor.middleware.limit$limit$fn__45621.invoke(limit.clj:38)"
metabase_1_c425448d3056 |   "query_processor.middleware.format_rows$format_rows$fn__45576.invoke(format_rows.clj:81)"
metabase_1_c425448d3056 |   "query_processor.middleware.desugar$desugar$fn__45050.invoke(desugar.clj:22)"
metabase_1_c425448d3056 |   "query_processor.middleware.binning$update_binning_strategy$fn__44076.invoke(binning.clj:229)"
metabase_1_c425448d3056 |   "query_processor.middleware.resolve_fields$resolve_fields$fn__44592.invoke(resolve_fields.clj:24)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_dimension_projections$add_remapping$fn__38811.invoke(add_dimension_projections.clj:318)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__39018.invoke(add_implicit_clauses.clj:141)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__39411.invoke(add_source_metadata.clj:105)"
metabase_1_c425448d3056 |   "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__46515.invoke(reconcile_breakout_and_order_by_bucketing.clj:98)"
metabase_1_c425448d3056 |   "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__43717.invoke(auto_bucket_datetimes.clj:125)"
metabase_1_c425448d3056 |   "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__44639.invoke(resolve_source_table.clj:46)"
metabase_1_c425448d3056 |   "query_processor.middleware.parameters$substitute_parameters$fn__46300.invoke(parameters.clj:114)"
metabase_1_c425448d3056 |   "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__44691.invoke(resolve_referenced.clj:80)"
metabase_1_c425448d3056 |   "query_processor.middleware.expand_macros$expand_macros$fn__45306.invoke(expand_macros.clj:158)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_timezone_info$add_timezone_info$fn__39442.invoke(add_timezone_info.clj:15)"
metabase_1_c425448d3056 |   "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__47212.invoke(splice_params_in_response.clj:32)"
metabase_1_c425448d3056 |   "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__46526$fn__46530.invoke(resolve_database_and_driver.clj:33)"
metabase_1_c425448d3056 |   "driver$do_with_driver.invokeStatic(driver.clj:61)"
metabase_1_c425448d3056 |   "driver$do_with_driver.invoke(driver.clj:57)"
metabase_1_c425448d3056 |   "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__46526.invoke(resolve_database_and_driver.clj:27)"
metabase_1_c425448d3056 |   "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__45524.invoke(fetch_source_query.clj:267)"
metabase_1_c425448d3056 |   "query_processor.middleware.store$initialize_store$fn__47221$fn__47222.invoke(store.clj:11)"
metabase_1_c425448d3056 |   "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
metabase_1_c425448d3056 |   "query_processor.store$do_with_store.invoke(store.clj:40)"
metabase_1_c425448d3056 |   "query_processor.middleware.store$initialize_store$fn__47221.invoke(store.clj:10)"
metabase_1_c425448d3056 |   "query_processor.middleware.cache$maybe_return_cached_results$fn__44568.invoke(cache.clj:209)"
metabase_1_c425448d3056 |   "query_processor.middleware.validate$validate_query$fn__47230.invoke(validate.clj:10)"
metabase_1_c425448d3056 |   "query_processor.middleware.normalize_query$normalize$fn__45648.invoke(normalize_query.clj:22)"
metabase_1_c425448d3056 |   "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__39280.invoke(add_rows_truncated.clj:36)"
metabase_1_c425448d3056 |   "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47197.invoke(results_metadata.clj:143)"
metabase_1_c425448d3056 |   "query_processor.middleware.constraints$add_default_userland_constraints$fn__44927.invoke(constraints.clj:42)"
metabase_1_c425448d3056 |   "query_processor.middleware.process_userland_query$process_userland_query$fn__46389.invoke(process_userland_query.clj:136)"
metabase_1_c425448d3056 |   "query_processor.middleware.catch_exceptions$catch_exceptions$fn__44870.invoke(catch_exceptions.clj:174)"
metabase_1_c425448d3056 |   "query_processor.reducible$async_qp$qp_STAR___38074$thunk__38075.invoke(reducible.clj:101)"
metabase_1_c425448d3056 |   "query_processor.reducible$async_qp$qp_STAR___38074$fn__38077.invoke(reducible.clj:106)"],
metabase_1_c425448d3056 |  :context :xlsx-download,
metabase_1_c425448d3056 |  :error "The maximum number of Cell Styles was exceeded. You can define up to 64000 style in a .xlsx Workbook",
metabase_1_c425448d3056 |  :row_count 0,
metabase_1_c425448d3056 |  :running_time 0,
metabase_1_c425448d3056 |  :preprocessed

```

**To Reproduce**
1. Go to Browse Data
2. Click on a report that contains ~100000 rows with DateTime columns
3. Export XLSX
4. Receive error

5. Open Editor
6. Click columns
7. Remove all DateTime columns
8. Close editor, refresh report
9. Export XLSX
10. Report OK

**Expected behavior**
The report should be exported also with DateTime columns

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "it-IT",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.83 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.8+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.8",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.8+10",
    "os.name": "Linux",
    "os.version": "5.4.0-47-generic",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.4"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-09-15",
      "tag": "v0.36.6",
      "branch": "release-0.36.x",
      "hash": "cb258fb"
    },
    "settings": {
      "report-timezone": "Europe/Berlin"
    }
  }
}
```

**Severity**
This bug was recently introduced. Upgrading to the latest version of the metabase is blocked because XLSX reports are "vital" for BI users to further manipulate data in Excel. On the other hand, the latest versions of Metabase correct too many errors and inaccuracies to make the upgrade urgent!

**Comment by flamber** [comment]
@d4rkstar Downgrade to 0.36.4
Related to #10803

**Comment by d4rkstar** [comment]
Hello @flamber , i tested 0.36.4 and it worked.

Will you leave this issue open until #10803 will be resolved?

**Comment by robdaemon** [comment]
Fixed by #13447
